packages/client/hmi-client/src/components/workflow/tera-simulate-chart.vue

```diff
@@ -228,8 +231,47 @@ const renderGraph = () => {
 					borderWidth: lineWidthArray.value[runIdx]
 				};
 				datasets.push(dataset);
-			})
-	);
+			});
+
+		if (props.initialData) {
+			let v = variable;
+			// get the dataset variable from the mapping of model variable to dataset variable if there's a mapping
+			if (props.mapping) {
+				if (!(props.mapping.length === 1 && Object.values(props.mapping[0]).some((val) => !val))) {
```

Comment:
I don't understand what you are checking in this predicate expression, especially the 2nd part
```
Object.values(props.mapping[0]).some((val) => !val)
```

Reply:
Checking for the case that where `props.mapping = [{datasetVariable: '', modelVariable: ''}]`. This is the case where a user has not provided an explicit mapping.

Reply:
Okay, can you add a code comment? because it isn't entirely obvious.